File: stacks/stack.py
class Stack:
  def __init__(self):
    self.stack = []

  def push(self, x):
    self.stack.append(x)

  def pop(self):
    self.stack.pop()

  def is_empty(self):
    return len(self.stack) == 0

  def peek(self):
    if self.is_empty():
      return None
    else:
      return self.stack[-1]

# test
my_stack = Stack()
print(my_stack.is_empty())
my_stack.push(10)
my_stack.push(3)
my_stack.push(-1)
my_stack.push(17)
print(my_stack.peek())
my_stack.pop()
print(my_stack.peek())
print(my_stack.is_empty())

File: stacks/stack.cpp
#include <iostream>

using namespace std;

#define MAX 1000

class stack {
  private:
    int top;
    int a[MAX];

  public:
    stack();
    void push(int);
    int pop();
    bool is_empty();
    int peek();
};

stack::stack() {
  top = -1;
}

void stack::push(int x) {
  a[++top] = x;
}

int stack::pop() {
  if(this->is_empty()) {
    cout << "Stack underflow!" << endl;
    return 0;
  } else {
    int x = a[top--];
    return x;
  }
}

bool stack::is_empty() {
  return top < 0;
}

int stack::peek() {
  if(this->is_empty()) {
    cout << "Stack is empty!" << endl;
    return 0;
  } else {
    return a[top];
  }
}


// test
int main() {
  stack my_stack;
  cout << my_stack.is_empty() << endl;
  my_stack.push(12);
  my_stack.push(-1);
  cout << my_stack.peek() << endl;
  my_stack.push(0);
  cout << my_stack.pop() <<  endl;
  cout << my_stack.pop() <<  endl;
  cout << my_stack.peek() << endl;
  cout << my_stack.is_empty() << endl;

  return 0;
}